### ISSUE TITLE
`vue/no-ref-as-operand` not catching `computed` refs

### ISSUE THREAD

**Issue by chetan** [vuejs/eslint-plugin-vue#2454]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.57.0
- **eslint-plugin-vue version:** 9.25.0
- **Vue version:** 3.4.24
- **Node version:** v20.12.1
- **Operating System:** macOS 14 (Sonoma)

**What did you do?**	

```vue
    const count = ref(0)
    const ok = ref(true)

    const notok = computed(() => !ok.value)
    if (notok || !ok) {
      console.log("not ok!!")
    }
```

**What did you expect to happen?**

`notok` should be flagged as a problem in the condition above.

**What actually happened?**

Only `!ok` is flagged.

It's happening in my local env but I am able to reproduce in the web docs:

<img width="788" alt="Screenshot 2024-04-22 at 11 38 42 AM" src="https://github.com/vuejs/eslint-plugin-vue/assets/25777/dab6f6b7-99e6-4f77-841e-d27d86da5b78">

**Repository to reproduce this issue**

can provide more info on my local configs if needed.

**Comment by waynzh** [comment]
For the error in the screenshot, you need to also import `computed` from vue.
![image](https://github.com/user-attachments/assets/1fcaecae-b216-4f1c-b4d0-fc0610dde8c6)


If you still have this error, please provide more information to help us debug this issue.

**Comment by waynzh** [comment]
I'm going to close this issue for now due to the lack of a reproduction, but feel free to add one if you have time and reopen this issue.
